docs/en/deploy/Running-Alluxio-On-Kubernetes.md

```diff
@@ -687,6 +684,136 @@ If Alluxio has already been deployed with helm and now you want to enable FUSE,
+
+#### Short-Circuit Modes
+There are 2 modes for using short-circuit.
+
+##### `local`
+In this mode, the Alluxio client and local Alluxio worker recognize each other if the client hostname 
+matches the worker hostname.
+This is called *Hostname Introspection*.
+In this mode, the Alluxio client and local Alluxio worker share the tiered storage of Alluxio worker.
+
+***Using `helm`***
+You can use `local` policy by setting the properties as below:
+
+```properties
+shortCircuit:
+  enabled: true
+  policy: local
+```
+
+***Using `kubectl`***
+In your `alluxio-configmap.yaml` you should add the following properties to `ALLUXIO_WORKER_JAVA_OPTS`:
+
+```properties
+-Dalluxio.user.short.circuit.enabled=true -Dalluxio.worker.data.server.domain.socket.address= -Dalluxio.worker.data.server.domain.socket.as.uuid=false
+```
+
+##### `uuid`
+This is the **default** policy used for short-circuit.
+
+If the client or worker container is using virtual networking, their hostnames may not match.
+In such a scenario, set the following property to use filesystem inspection to enable short-circuit
+operations and **make sure the client container mounts the directory specified as the domain socket
+path**.
+Short-circuit writes are then enabled if the worker UUID is located on the client filesystem.
+
+***Domain Socket Path***
+The domain socket is a volume which should be mounted on:
+
+- All Alluxio workers
+- All application containers which intend to read/write through Alluxio
+
+This domain socket volume refers to a `PersistentVolumeClaim` by its name.
+You should provision to the `PersistenceVolumeClaim` with a `PersistentVolume` of kind being either
+`hostPath` or `local`.
+
+***Using `helm`***
+You can use `uuid` policy by setting the properties as below:
+
+```properties
+# These are the default configurations
+shortCircuit:
+  enabled: true
+  policy: uuid
+  pvcName: alluxio-worker-domain-socket
+  accessModes:
+    - ReadWriteOnce
+  size: 1Gi
+  storageClass: standard
+```
+
+The field `shortCircuit.pvcName` defines the name of the `PersistentVolumeClaim` for domain socket.
+
+***Using `kubectl`***
+You should verify the following properties in `ALLUXIO_WORKER_JAVA_OPTS`.
+Actually they are set to these values by default:
+```properties
+-Dalluxio.worker.data.server.domain.socket.address=/opt/domain -Dalluxio.worker.data.server.domain.socket.as.uuid=true
+```
+
+Also you should make sure the worker Pods have domain socket defined in the `volumes`,
+and all relevant containers have the domain socket volume mounted.
+The domain socket volume is defined as below by default:
+```properties
+volumes:
+  - name: alluxio-domain
+    persistentVolumeClaim:
+      claimName: "alluxio-worker-domain-socket"
+```
+
+> Note: Compute application containers **MUST** mount the domain socket volume to the same path
+(`/opt/domain`) as configured for the Alluxio workers.
+
+The `PersistenceVolumeClaim` is defined in `worker/alluxio-worker-pvc.yaml`.
```

Comment:
is this a helm template path? the user does not need to know about this. when using `kubectl` one of the pre-generated templates would be used.

Reply:
This is a pre-generated template path. (I'm adding `.template` to the end to make it more clear)